atsign-foundation/at_client_sdk
# fix: Add value validations

**- What I did**
* Fixes the git issue: https://github.com/atsign-foundation/at_client_sdk/issues/923

**- How I did it**
* Added IF conditions to check if the value size exceeds the buffer limit for put and notify methods.

**- How to verify it**
* Added unit test to verify put methods throws an exception when value exceeds the maximum buffer limit.
* Added unit test to verify notify method throws an exception when the value exceeds the maximum buffer limit.

TODO: Refactor the notify method in the "notification_service_impl.dart" to pass the mock object of the "NotificationRequestTransformer" instance and assert the buffer overflow exception is caught and propagated back to the client. Prefer making these changes in a separate PR.

**- Description for the changelog**
* Added condition to prevent value exceeding the buffer limit.
